hathor/indexes/addresses_memory_index.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+from typing import TYPE_CHECKING, DefaultDict, Iterable, List, Optional, Set
+
+from structlog import get_logger
+
+from hathor.indexes.addresses_index import AddressesIndex
+from hathor.pubsub import HathorEvents
+from hathor.transaction import BaseTransaction
+from hathor.transaction.scripts import parse_address_script
+
+if TYPE_CHECKING:  # pragma: no cover
+    from hathor.pubsub import EventArguments, PubSubManager
+    from hathor.transaction import TxOutput
+
+logger = get_logger()
+
+
+class AddressesMemoryIndex(AddressesIndex):
```

Comment:
I'm not sure about the infix these names, also maybe just "address index" instead of "addresses index", and instead of the infix, a prefix, like the managers: "rocksdb address index", "memory address index".